### PR TITLE
Optimizing and fixing mistakes from Milestone 3

### PR DESCRIPTION
**Pull request by DawnTheWitch**
There are some additional bug fixes that were fixed from these. However the strange extra atoms popping up is fixed. Issues #20 and #21 are still being worked on

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
  * @author Dawn Moore
  * @author James Oswald
  */
 
-import {AEGTree} from "./AEG/AEGTree";
-import {Point} from "./AEG/Point";
-import {CutNode} from "./AEG/CutNode";
-import {Ellipse} from "./AEG/Ellipse";
-import {AtomNode} from "./AEG/AtomNode";
+import "./index.css";
```
Comment (James-Oswald) [inline src/index.ts:14]: This CSS shouldn't be necessary anymore. I got rid of this when we swapped to Vite, which can actually detect the includes from the HTML.
```suggestion

```

File: src/index.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+import {Ellipse} from "./AEG/Ellipse";
+import {AtomNode} from "./AEG/AtomNode";
+import {ellipseCreation, removeCutListener} from "./EllipseCreation";
+import {atomCreation, removeAtomListener} from "./AtomCreation";
+
 /**
  * A program to draw ellipses and atoms.
  * @author Dawn Moore
  * @author James Oswald
  */
```
Comment (James-Oswald) [inline src/index.ts:13]: Keep file comment above imports.  
```suggestion
/**
 * A program to draw ellipses and atoms.
 * @author Dawn Moore
 * @author James Oswald
 */

import {AEGTree} from "./AEG/AEGTree";
import {CutNode} from "./AEG/CutNode";
import {Ellipse} from "./AEG/Ellipse";
import {AtomNode} from "./AEG/AtomNode";
import {ellipseCreation, removeCutListener} from "./EllipseCreation";
import {atomCreation, removeAtomListener} from "./AtomCreation";
```

**Review by James-Oswald — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (DawnTheWitch) [reply src/index.ts:14]: Yeah I thought so too, but when I took it out this is what the HTML turned into

d60cd47022f5a3ecf4110e10a043aff1

File: src/EllipseCreation.ts [hunk 3]
```diff
@@ -55,44 +57,64 @@ function drawEllipse(original: Point, current: Point) {
...
+/**
+ * When the mouse is lifted up, removes the movement listener and adds it to the tree itself.
+ */
+function mouseUp() {
+    canvas.removeEventListener("mousemove", mouseMoving);
+    const newCut: CutNode = new CutNode(currentEllipse);
+    if (tree.canInsertAEG(newCut, currentEllipse.center)) {
+        tree.insertAEG(newCut, currentEllipse.center);
+    }
+    canvas.removeEventListener("mouseup", mouseUp);
```
Comment (James-Oswald) [inline src/EllipseCreation.ts:112]: Id prefer you keep the together so its more clear. 
```suggestion
    canvas.removeEventListener("mousemove", mouseMoving);
    canvas.removeEventListener("mouseup", mouseUp);
    const newCut: CutNode = new CutNode(currentEllipse);
    if (tree.canInsertAEG(newCut, currentEllipse.center)) {
        tree.insertAEG(newCut, currentEllipse.center);
    }
```

File: src/index.ts (hunk 1 above)
Comment (James-Oswald) [reply src/index.ts]: Resolved via `npx vite preview` / `npm run live-edit`

**Review by James-Oswald — APPROVED** [review]

**Review by James-Oswald — APPROVED** [review]

**Pull request merged**